### PR TITLE
http-gateway.md: create deployment doc based on yaml config.

### PR DESCRIPTION
**Pull request by Trishia**
@ondrejtomcik
I newly created this guide based on yaml config as well as making certificates, getting/editing config file, running image without source download & build. Review and then give me comments. If good, approve plz.

### CODE REVIEW

File: docs/guide/deployment/http-gateway.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+	-e CLOUD_SID=00000000-0000-0000-0000-000000000001 \
+	--entrypoint /bin/bash \
+	plgd/bundle:latest
+
+# Copy & paste below commands on the bash shell of plgd/bundle container.
+certificate-generator --cmd.generateRootCA --outCert=/certs/root_ca.crt --outKey=/certs/root_ca.key --cert.subject.cn=RootCA
+certificate-generator --cmd.generateCertificate --outCert=/certs/http.crt --outKey=/certs/http.key --cert.subject.cn=localhost --cert.san.domain=localhost --signerCert=/certs/root_ca.crt --signerKey=/certs/root_ca.key
+certificate-generator --cmd.generateIdentityCertificate=$CLOUD_SID --outCert=/certs/coap.crt --outKey=/certs/coap.key --cert.san.domain=localhost --signerCert=/certs/root_ca.crt --signerKey=/certs/root_ca.key
+cat /certs/http.crt > /certs/mongo.key
+cat /certs/http.key >> /certs/mongo.key
```
Comment (jkralik) [inline docs/guide/deployment/http-gateway.md]: These steps are needed only for run mongo DB . Any our services doesn't need it. Pls could you remove them from all other services.

File: docs/guide/deployment/http-gateway.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+	--network=host \
+	-v $(pwd)/.tmp/certs:/certs \
+	-e CLOUD_SID=00000000-0000-0000-0000-000000000001 \
+	--entrypoint /bin/bash \
+	plgd/bundle:latest
+
+# Copy & paste below commands on the bash shell of plgd/bundle container.
+certificate-generator --cmd.generateRootCA --outCert=/certs/root_ca.crt --outKey=/certs/root_ca.key --cert.subject.cn=RootCA
+certificate-generator --cmd.generateCertificate --outCert=/certs/http.crt --outKey=/certs/http.key --cert.subject.cn=localhost --cert.san.domain=localhost --signerCert=/certs/root_ca.crt --signerKey=/certs/root_ca.key
+certificate-generator --cmd.generateIdentityCertificate=$CLOUD_SID --outCert=/certs/coap.crt --outKey=/certs/coap.key --cert.san.domain=localhost --signerCert=/certs/root_ca.crt --signerKey=/certs/root_ca.key
```
Comment (jkralik) [inline docs/guide/deployment/http-gateway.md]: This step is needed only for coap-gateway.

**Review by jkralik — COMMENTED** [review]
Pls check my comments.

File: docs/guide/deployment/http-gateway.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
+# HTTP gateway
+
```
Comment (jkralik) [inline docs/guide/deployment/http-gateway.md:3]: Pls add short description.

BTW: http-gateway exposes REST API and websockets for events.

**Review by jkralik — COMMENTED** [review]
Pls add point to swagger definition for REST API: https://petstore.swagger.io/?url=https://raw.githubusercontent.com/plgd-dev/cloud/v2/http-gateway/swagger.yaml

**Comment by jkralik** [comment]
Pls remove http-gateway/README.md and create symlink to docs/guide/deployment/http-gateway.md

```bash
cd  http-gateway
rm README.md
ln -sr ../docs/guide/deployment/http-gateway.md README.md
```

**Comment by ondrejtomcik** [comment]
> Pls add point to swagger definition for REST API: https://petstore.swagger.io/?url=https://raw.githubusercontent.com/plgd-dev/cloud/v2/http-gateway/swagger.yaml

This is @jkralik deployment configuration. Link to API will be on a different place.

**Comment by Trishia** [comment]
> > Pls add point to swagger definition for REST API: https://petstore.swagger.io/?url=https://raw.githubusercontent.com/plgd-dev/cloud/v2/http-gateway/swagger.yaml
> 
> This is @jkralik deployment configuration. Link to API will be on a different place.

I mentioned swagger link in '### HTTP API` section of Configuration.

File: docs/guide/deployment/http-gateway.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
+# HTTP gateway
```
Comment (ondrejtomcik) [inline docs/guide/deployment/http-gateway.md]: HTTP Gateway

File: docs/guide/deployment/http-gateway.md [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+Http gateway exposes REST API for devices, cloud configuration, certificate authority and websockets for events as well as user web UI (i.e. PLGD Dashboard).
```
Comment (ondrejtomcik) [inline docs/guide/deployment/http-gateway.md]: HTTP Gateway exposes the client's [REST API](swagger...) to manage the user's devices, as well as the Web UI known as [plgd Dashboard](... for now leave three dots, will update later).

File: docs/guide/deployment/http-gateway.md [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+| `clients.certificateAuthority.grpc.tls.keyFile` | string | `File path to private key in PEM format.` | `""` |
+| `clients.certificateAuthority.grpc.tls.certFile` | string | `File path to certificate in PEM format.` | `""` |
+| `clients.certificateAuthority.grpc.tls.useSystemCAPool` | bool | `If true, use system certification pool.` | `false` |
+| `clients.certificateAuthority.grpc.keepAlive.time` | string | `After a duration of this time if the client doesn't see any activity it pings the server to see if the transport is still alive.` | `10s` |
+| `clients.certificateAuthority.grpc.keepAlive.timeout` | string | `After having pinged for keepalive check, the client waits for a duration of Timeout and if no activity is seen even after that the connection is closed.` | `20s` |
+| `clients.certificateAuthority.grpc.keepAlive.permitWithoutStream` | bool | `If true, client sends keepalive pings even with no active RPCs. If false, when there are no active RPCs, Time and Timeout will be ignored and no keepalive pings will be sent.` | `false` |
+
+### User Web UI
```
Comment (ondrejtomcik) [inline docs/guide/deployment/http-gateway.md]: Just Web UI

**Review by ondrejtomcik — APPROVED** [review]

**Pull request merged**